spec/controllers/user_journey_controller_spec.rb

```diff
@@ -296,4 +280,32 @@
       expect(flash[:error]).to include(I18n.t('certificates.errors.invalid'))
     end
   end
+
+  context 'Enforcement of component team restrictions' do
+    it 'prevents rendering of the upload certificate page when team is not matching' do
+      certmgr_stub_auth
+      foreign_team = FactoryBot.create(:team, id: SecureRandom.uuid)
+      foreign_component = FactoryBot.create(:sp_component, team_id: foreign_team.id)
+      get :upload_certificate,
+          params: {
+            component_type: foreign_component.component_type,
+            component_id: foreign_component.id,
+            certificate_id: msa_encryption_cert.id
+          }
+      expect(response).to_not have_http_status(:success)
+      expect(subject).to_not render_template(:upload_certificate)
+      expect(response).to have_http_status(:forbidden)
+      expect(flash[:warn]).to eq t('shared.errors.authorisation')
+    end
+
+    it 'behaves gracefully if a non-existent component is accessed' do
+      certmgr_stub_auth
+      get :upload_certificate,
+          params: {
+            component_type: :sp_component,
+            component_id: 'bad id',
+            certificate_id: msa_encryption_cert.id
+          }
+    end
```

Comment:
Not so keen on not having any expectations really. Can we test that that we are not rendering the template that would have been rendered if we didn't have the above guard?

Reply:
It does get rendered; the guard isn't supposed to be preventing the rendering, it's there to avoid having the authorisation code blow up.  Proper handling of trying to navigate to a non-existent entity is a separate concern, and is beyond the scope of this task.  We don't just need to fix it for components, we need to fix it for everything.  It just happens that there's additionally an authorisation concern with components.

Reply:
in that case, use `expect(subject).to render_template(:upload_certificate)` ... that proves the authorisation code is being graceful and doesn't halt the execution/rendering... a test without an expectation isn't really a test...

Reply:
Righty-ho, done.  We did agree yesterday that it wouldn't have any though.

The advantage of adding one is that it'll break when we do add handling for navigation to non-existent entities, which will prompt us to review the expectations on the test, rather than running the risk of it getting left.

Reply:
Sorry my bad, I meant not to have a meaningful expectation that everything works, i.e. not proving it works e2e but the authorization doesn't break stuff...